render_silhouette_regmarks.py

```diff
@@ -0,0 +1,168 @@
+"""
+
+import inkex
+from lxml import etree
+from gettext import gettext
+
+SVG_URI = u'http://www.w3.org/2000/svg'
+
+class InsertRegmark(inkex.Effect):
+	def __init__(self):
+		inkex.Effect.__init__(self)
+		
+		# Layer name static, since self.document.getroot() not available on initialization
+		self.layername = 'Regmarks'
+		
+		# Parse arguments
+		self.arg_parser.add_argument("-X", "--reg-x", "--regwidth", 
+				type = float, dest = "regwidth", default = 180.0, help="X mark distance [mm]")
+		self.arg_parser.add_argument("-Y", "--reg-y", "--reglength", 
+				type = float, dest = "reglength", default = 230.0, help="Y mark distance [mm]")
+		self.arg_parser.add_argument("--rego-x",  "--regoriginx", 
+				type = float, dest = "regoriginx", default = 15.0, help="X mark origin from left [mm]")
+		self.arg_parser.add_argument("--rego-y", "--regoriginy", 
+				type = float, dest = "regoriginy", default = 20.0, help="X mark origin from top [mm]")
+		self.arg_parser.add_argument("--verbose", dest = "verbose", type = inkex.Boolean, default = False, help="enable log messages")
+
+	#SVG rect element generation routine
+	def drawRect(self, size, pos, name):
+		x, y = [pos * self.svg.unittouu('1mm') for pos in pos  ]
+		w, h = [pos * self.svg.unittouu('1mm') for pos in size ]
+		rect = etree.Element('{%s}rect' % SVG_URI)
+		rect.set('x', str(x))
+		rect.set('y', str(y))
+		rect.set('id', name)
+		rect.set('width', str(w))
+		rect.set('height', str(h))
+		rect.set('style', 'fill: black;')
+		return rect
+		
+	#SVG line element generation routine
+	def drawLine(self, posStart, posEnd, name):
+		x1, y1, = [pos * self.svg.unittouu('1mm') for pos in posStart]
+		x2, y2, = [pos * self.svg.unittouu('1mm') for pos in posEnd  ]
+		line = etree.Element('{%s}line' % SVG_URI)
+		line.set('x1', str(x1))
+		line.set('y1', str(y1))
+		line.set('x2', str(x2))
+		line.set('y2', str(y2))
+		line.set('id', name)
+		# https://www.reddit.com/r/silhouettecutters/comments/wcdnzy/the_key_to_print_and_cut_success_an_extensive/
+		# > The registration mark thickness is actually very important. For some reason, 0.3 mm marks work perfectly. 
+		# > The thicker you get, the less accurate registration will be. ~~~ galaxyman47
+		REG_MARK_LINE_WIDTH_MM = 0.3
+		line.set('style', 'stroke: black; stroke-width: '+str(REG_MARK_LINE_WIDTH_MM* self.svg.unittouu('1mm'))+';')
+		return line
+	
+	def effect(self):
+		REG_SQUARE_MM = 5
+		REG_LINE_MM = 20
+		REG_KEEPOUT_MM = 2
+
+		svg = self.document.getroot()
+
+		reg_origin_X = self.options.regoriginx
+		reg_origin_Y = self.options.regoriginy
+		reg_width = self.options.regwidth if self.options.regwidth else int(svg.get("width").rstrip("mm")) - reg_origin_X*2
+		reg_length = self.options.reglength if self.options.reglength else int(svg.get("height").rstrip("mm")) - reg_origin_Y*2
+
+		if self.options.verbose == True:
+			inkex.errormsg(gettext("[INFO]: page width ")+str(svg.get("width").rstrip("mm")))
```

Comment:
curious: would this also work with `inkex.msg()` ?

https://inkscape.gitlab.io/extensions/documentation/source/inkex.base.html?highlight=msg#inkex.base.InkscapeExtension.msg